### PR TITLE
Add overloaded method contains of AbstractDoubleArrayAssert and AbstractFloatArrayAssert

### LINKED ISSUE #2055: Add overloaded methods with boxed arrays for contains assertions
#### Summary
Related to #1948, we should add the overloaded methods with boxed arrays for all the primitive array types:

- [x] `AbstractBooleanArrayAssert` (84fc616)
- [x] `AbstractByteArrayAssert` (6936b6133eec1f6a11e052724e69bd77ed0b19ab)
- [x] `AbstractCharArrayAssert` (48794650806b33dbfd8f72f2fc09bf7d92686d0b)
- [x] `AbstractDoubleArrayAssert` (dcfdd05dd01030bcd6f6210d4d48455d49c51b52)
- [x] `AbstractFloatArrayAssert` (dcfdd05dd01030bcd6f6210d4d48455d49c51b52)
- [x] `AbstractIntArrayAssert` (0c5f366dcfd31dee9867ccafbec7dc8a97f5d100)
- [x] `AbstractLongArrayAssert` (#2042)
- [x] `AbstractShortArrayAssert` (6936b6133eec1f6a11e052724e69bd77ed0b19ab)

### PR DESCRIPTION
**Pull request by OmarMorales71**
#### Check List:
* Fixes two elements of #2055 
* Unit tests : YES
* Javadoc with a code example (on API only) : YES

### CODE REVIEW

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java [hunk 1]
```diff
@@ -224,6 +226,33 @@ public SELF contains(double... values) {
...
+   *
+   * @param values the given values.
+   * @return {@code this} assertion object.
+   * @throws NullPointerException if the given argument is {@code null}.
+   * @throws IllegalArgumentException if the given argument is an empty array.
+   * @throws AssertionError if the actual array is {@code null}.
+   * @throws AssertionError if the actual array does not contain the given values.
+   * @since 3.19.0
+   */
+  public SELF contains(Double[] values) {
```
Comment (scordio) [inline src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:250]: Please also add 
```java
public SELF contains(Double[] values, Offset<Double> precision)
```

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java [hunk 2]
```diff
@@ -897,6 +1098,31 @@ public SELF containsExactly(double... values) {
...
+   * @param values the given values.
+   * @return {@code this} assertion object.
+   * @throws NullPointerException if the given argument is {@code null}.
+   * @throws AssertionError if the actual group is {@code null}.
+   * @throws AssertionError if the actual group does not contain the given values with same order, i.e. the actual group
+   *           contains some or none of the given values, or the actual group contains more values than the given ones
+   *           or values are the same but the order is not.
+   * @since 3.19.0
+   */
+  public SELF containsExactly(Double[] values) {
```
Comment (scordio) [inline src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:1120]: Please also add
```java
public SELF containsExactly(Double[] values, Offset<Double> precision)
```

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java [hunk 3]
```diff
@@ -285,6 +314,33 @@ public SELF containsOnly(double... values) {
...
+   *
+   * @param values the given values.
+   * @return {@code this} assertion object.
+   * @throws NullPointerException if the given argument is {@code null}.
+   * @throws IllegalArgumentException if the given argument is an empty array.
+   * @throws AssertionError if the actual array is {@code null}.
+   * @throws AssertionError if the actual array does not contain the given values, i.e. the actual array contains some
+   *           or none of the given values, or the actual array contains more values than the given ones.
+   */
+  public SELF containsOnly(Double[] values) {
```
Comment (scordio) [inline src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:338]: Please also add
```java
public SELF containsOnly(Double[] values, Offset<Double> precision)
```

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java [hunk 4]
```diff
@@ -345,6 +401,32 @@ public SELF containsOnlyOnce(double... values) {
...
+   *
+   * @param values the given values.
+   * @return {@code this} assertion object.
+   * @throws NullPointerException if the given argument is {@code null}.
+   * @throws IllegalArgumentException if the given argument is an empty array.
+   * @throws AssertionError if the actual array is {@code null}.
+   * @throws AssertionError if the actual group does not contain the given values, i.e. the actual group contains some
+   *           or none of the given values, or the actual group contains more than once these values.
+   */
+  public SELF containsOnlyOnce(Double[] values) {
```
Comment (scordio) [inline src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:424]: Please also add
```java
public SELF containsOnlyOnce(Double[] values, Offset<Double> precision)
```

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java [hunk 5]
```diff
@@ -404,6 +486,29 @@ public SELF containsSequence(double... sequence) {
...
+   * // assertion will fail
+   * assertThat(new double[] { 1.0, 1.0, 2.0 }).containsSequence(new Double[] {2.0, 1.0});</code></pre>
+   *
+   * @param sequence the sequence of values to look for.
+   * @return myself assertion object.
+   * @throws AssertionError if the actual array is {@code null}.
+   * @throws AssertionError if the given array is {@code null}.
+   * @throws AssertionError if the actual array does not contain the given sequence.
+   */
+  public SELF containsSequence(Double[] sequence) {
```
Comment (scordio) [inline src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:506]: Please also add
```java
public SELF containsSequence(Double[] sequence, Offset<Double> precision)
```

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java [hunk 6]
```diff
@@ -464,6 +569,29 @@ public SELF containsSubsequence(double... subsequence) {
...
+   * // assertion will fail
+   * assertThat(new double[] { 1.0, 1.0, 2.0 }).containsSubsequence(new Double[] { 2.0, 1.0 });</code></pre>
+   *
+   * @param subsequence the subsequence of values to look for.
+   * @return myself assertion object.
+   * @throws AssertionError if the actual array is {@code null}.
+   * @throws AssertionError if the given array is {@code null}.
+   * @throws AssertionError if the actual array does not contain the given subsequence.
+   */
+  public SELF containsSubsequence(Double[] subsequence) {
```
Comment (scordio) [inline src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:589]: Please also add
```java
public SELF containsSubsequence(Double[] subsequence, Offset<Double> precision)
```

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java [hunk 7]
```diff
@@ -586,6 +714,29 @@ public SELF doesNotContain(double... values) {
...
+   * assertThat(new double[] { 1.0, 1.0, 2.0 }).doesNotContain(new Double[] { 2.0 });</code></pre>
+   *
+   * @param values the given values.
+   * @return {@code this} assertion object.
+   * @throws NullPointerException if the given argument is {@code null}.
+   * @throws IllegalArgumentException if the given argument is an empty array.
+   * @throws AssertionError if the actual array is {@code null}.
+   * @throws AssertionError if the actual array contains any of the given values.
+   */
+  public SELF doesNotContain(Double[] values) {
```
Comment (scordio) [inline src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:734]: Please also add

```java
public SELF doesNotContain(Double[] values, Offset<Double> precision)
```

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java [hunk 8]
```diff
@@ -747,6 +898,31 @@ public SELF startsWith(double... sequence) {
...
+   * assertThat(new double[] { 1.0, 2.0, 2.0, 1.0 }).startsWith(new Double[] {2.0, 2.0});</code></pre>
+   *
+   * @param sequence the sequence of values to look for.
+   * @return myself assertion object.
+   * @throws NullPointerException if the given argument is {@code null}.
+   * @throws IllegalArgumentException if the given argument is an empty array.
+   * @throws AssertionError if the actual array is {@code null}.
+   * @throws AssertionError if the actual array does not start with the given sequence.
+   */
+  public SELF startsWith(Double[] sequence) {
```
Comment (scordio) [inline src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:920]: Please also add
```java
public SELF startsWith(Double[] sequence, Offset<Double> precision)
```

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java [hunk 9]
```diff
@@ -807,6 +983,31 @@ public SELF endsWith(double... sequence) {
...
+   * assertThat(new double[] { 1.0, 2.0, 2.0, 1.0 }).endsWith(new Double[] {1.0, 2.0});</code></pre>
+   *
+   * @param sequence the sequence of values to look for.
+   * @return myself assertion object.
+   * @throws NullPointerException if the given argument is {@code null}.
+   * @throws IllegalArgumentException if the given argument is an empty array.
+   * @throws AssertionError if the actual array is {@code null}.
+   * @throws AssertionError if the actual array does not end with the given sequence.
+   */
+  public SELF endsWith(Double[] sequence) {
```
Comment (scordio) [inline src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:1005]: Please also add
```java
public SELF endsWith(Double[] sequence, Offset<Double> precision)
```

File: src/main/java/org/assertj/core/api/AbstractFloatArrayAssert.java [hunk 10]
```diff
@@ -223,6 +225,32 @@ public SELF contains(float... values) {
...
+   *
+   * @param values the given values.
+   * @return {@code this} assertion object.
+   * @throws NullPointerException if the given argument is {@code null}.
+   * @throws IllegalArgumentException if the given argument is an empty array.
+   * @throws AssertionError if the actual array is {@code null}.
+   * @throws AssertionError if the actual array does not contain the given values.
+   * @since 3.19.0
+   */
+  public SELF contains(Float[] values) {
```
Comment (scordio) [inline src/main/java/org/assertj/core/api/AbstractFloatArrayAssert.java:248]: Please add the corresponding variants for all the `containsXXX(double[], Offset<Float>)` methods, e.g.:

```java
public SELF contains(Float[] values, Offset<Float> precision)
```

File: src/main/java/org/assertj/core/api/AbstractFloatArrayAssert.java [hunk 11]
```diff
@@ -746,6 +896,31 @@ public SELF startsWith(float... sequence) {
...
   }
 
+  /**
+   * Verifies that the actual array starts with the given sequence of values, without any other values between them.
+   * Similar to <code>{@link #containsSequence(Float[])}</code>, but it also verifies that the first element in the
+   * sequence is also first element of the actual array.
+   * <p>
+   * Example:
+   * <pre><code class='java'> // assertion will pass
+   * assertThat(new float[] { 1.0f, 2.0f, 3.0f, 4.0f }).startsWith(new Float[] {1.0f, 2.0f});
```
Comment (scordio) [inline src/main/java/org/assertj/core/api/AbstractFloatArrayAssert.java:906]:
```suggestion
   * assertThat(new float[] { 1.0f, 2.0f, 3.0f, 4.0f }).startsWith(new Float[] { 1.0f, 2.0f });
```

File: src/main/java/org/assertj/core/api/AbstractFloatArrayAssert.java [hunk 12]
```diff
@@ -463,6 +567,29 @@ public SELF containsSubsequence(float... subsequence) {
     return myself;
   }
 
+  /**
+   * Verifies that the actual array contains the given subsequence (possibly with other values between them).
+   * <p>
+   * Example:
+   * <pre><code class='java'> // assertion will pass
+   * assertThat(new float[] { 1.0f, 2.0f }).containsSubsequence(new Float[] { 1.0f, 2.0f});
```
Comment (scordio) [inline src/main/java/org/assertj/core/api/AbstractFloatArrayAssert.java:575]:
```suggestion
   * assertThat(new float[] { 1.0f, 2.0f }).containsSubsequence(new Float[] { 1.0f, 2.0f });
```

**Review by scordio — CHANGES_REQUESTED** [review]
Thank you @OmarMorales71, some points to fix and then it's ready. It's mostly about keeping the javadoc style consistent and adding additional methods for the variants with precision.

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java (hunk 1 above)
Comment (OmarMorales71) [reply src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:249]: https://github.com/assertj/assertj-core/blob/c0f48d3858e047ed0421a36c435e6f1b5503edcb/src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java#L249

It's already there

Comment (OmarMorales71) [reply src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:249]: https://github.com/assertj/assertj-core/blob/c0f48d3858e047ed0421a36c435e6f1b5503edcb/src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java#L249

It's already there

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java (hunk 2 above)
Comment (OmarMorales71) [reply src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:1120]: https://github.com/assertj/assertj-core/blob/c0f48d3858e047ed0421a36c435e6f1b5503edcb/src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java#L951

It's already there

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java (hunk 3 above)
Comment (OmarMorales71) [reply src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:338]: https://github.com/assertj/assertj-core/blob/c0f48d3858e047ed0421a36c435e6f1b5503edcb/src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java#L312

It's already there

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java (hunk 4 above)
Comment (OmarMorales71) [reply src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:424]: https://github.com/assertj/assertj-core/blob/c0f48d3858e047ed0421a36c435e6f1b5503edcb/src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java#L370

It's already there

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java (hunk 6 above)
Comment (OmarMorales71) [reply src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:589]: https://github.com/assertj/assertj-core/blob/c0f48d3858e047ed0421a36c435e6f1b5503edcb/src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java#L430

It's already there

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java (hunk 5 above)
Comment (OmarMorales71) [reply src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:506]: https://github.com/assertj/assertj-core/blob/c0f48d3858e047ed0421a36c435e6f1b5503edcb/src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java#L430

It's already there

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java (hunk 6 above)
Comment (OmarMorales71) [reply src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:589]: https://github.com/assertj/assertj-core/blob/c0f48d3858e047ed0421a36c435e6f1b5503edcb/src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java#L490

It's already there

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java (hunk 7 above)
Comment (OmarMorales71) [reply src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:734]: https://github.com/assertj/assertj-core/blob/c0f48d3858e047ed0421a36c435e6f1b5503edcb/src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java#L610

It's already there

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java (hunk 8 above)
Comment (OmarMorales71) [reply src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:920]: https://github.com/assertj/assertj-core/blob/c0f48d3858e047ed0421a36c435e6f1b5503edcb/src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java#L775

It's already there

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java (hunk 9 above)
Comment (OmarMorales71) [reply src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:1005]: https://github.com/assertj/assertj-core/blob/c0f48d3858e047ed0421a36c435e6f1b5503edcb/src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java#L835

It's already there

File: src/main/java/org/assertj/core/api/AbstractFloatArrayAssert.java (hunk 10 above)
Comment (OmarMorales71) [reply src/main/java/org/assertj/core/api/AbstractFloatArrayAssert.java:248]: https://github.com/assertj/assertj-core/blob/c0f48d3858e047ed0421a36c435e6f1b5503edcb/src/main/java/org/assertj/core/api/AbstractFloatArrayAssert.java#L248

It's already there. Also the rest of the methods you're talking about.

**Comment by OmarMorales71** [comment]
@scordio I made the changes. I think the PR is ready. Let me know if something is missing

File: src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java (hunk 1 above)
Comment (scordio) [reply src/main/java/org/assertj/core/api/AbstractDoubleArrayAssert.java:249]: That one is with primitive `double[]` array as first parameter, we would need the boxed `Double[]` array

**Comment by OmarMorales71** [comment]
@scordio I created the new methods and added unit tests for those methods.

**Comment by scordio** [comment]
@OmarMorales71 thank you! I'll have a look soon.

**Comment by scordio** [comment]
Integrated, thanks @OmarMorales71!

**Pull request closed without merging**